File: Java/0557_Reverse_Words_in_a_String_III.java
class Solution {
    public String reverseWords(String s) {
        String[] words = s.split(" ");
        String[] result = new String[words.length];
        for(int i = 0; i < words.length; i++){
            result[i] = reverse(words[i]);
        }
        return String.join(" ", result);
    }
    private String reverse(String s){
        char[] cArray = s.toCharArray();
        int n = cArray.length;;
        char temp;
        for(int i = 0; i < n/2; i++){
            temp = cArray[i];
            cArray[i] = cArray[n-1-i];
            cArray[n-1-i] = temp;
        }
        return new String(cArray);
        /*
        StringBuilder sb = new StringBuilder(s);
        sb.reverse();
        return sb.toString();
        */
    }
}

File: CS/0557_Reverse_Words_in_a_String_III.cs
public class Solution {
    public string ReverseWords(string s) {
        StringBuilder result = new StringBuilder();
        string[] sArr = s.Split(" ");
        for(int i = 0; i < sArr.Length; i++){
            for(int j = sArr[i].Length-1; j >= 0; j--){
                result.Append(sArr[i][j]);
            }
            result.Append( i != sArr.Length-1 ? " " : "");
        }
        
        return result.ToString();
    }
}
